cmd/batch-expire.go

```diff
@@ -538,15 +540,34 @@ func (r *BatchJobExpire) Start(ctx context.Context, api ObjectLayer, job BatchJo
 	defer cancel()
 
 	results := make(chan itemOrErr[ObjectInfo], workerSize)
-	if err := api.Walk(ctx, r.Bucket, r.Prefix, results, WalkOptions{
-		Marker:       lastObject,
-		LatestOnly:   false, // we need to visit all versions of the object to implement purge: retainVersions
-		VersionsSort: WalkVersionsSortDesc,
-	}); err != nil {
-		// Do not need to retry if we can't list objects on source.
-		return err
+	prefixs := strings.Split(r.Prefix, ",")
+	wg := sync.WaitGroup{}
+	for _, prefix := range prefixs {
+		prefixResultCh := make(chan itemOrErr[ObjectInfo], workerSize)
+		err := api.Walk(ctx, r.Bucket, strings.TrimSpace(prefix), prefixResultCh, WalkOptions{
+			Marker:       lastObject,
+			LatestOnly:   false, // we need to visit all versions of the object to implement purge: retainVersions
+			VersionsSort: WalkVersionsSortDesc,
+		})
+		if err != nil {
+			xioutil.SafeClose(results)
+			// Do not need to retry if we can't list objects on source.
+			return err
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for result := range prefixResultCh {
+				results <- result
+			}
+		}()
 	}
 
+	go func() {
+		wg.Wait()
+		xioutil.SafeClose(results)
+	}()
```

Comment:
We don't need to list them in parallel, we should list them individually in the specified order of entries.